x-pack/plugins/dataset_quality/public/hooks/use_dataset_quality_table.tsx

```diff
@@ -63,10 +76,18 @@ export const useDatasetQualityTable = () => {
 
   const renderedItems = useMemo(() => {
     const overridenSortingField = sortingOverrides[sortField] || sortField;
-    const sortedItems = orderBy(data, overridenSortingField, sortDirection);
+    const mergedData = data.map((dataStream) => {
+      const malformedDocs = find(malformedStats, { dataset: dataStream.name });
+
+      return malformedDocs
+        ? merge({}, dataStream, { malformedDocs: malformedDocs.percentage })
```

Comment:
afaik main difference is `merge` is able to recursively merges properties at every level whilst `Object.assign` will drop anything that is not top level from the `dataStream` object

Reply:
In this case should still be good Object.assign, since it basically add a primitive property to the dataStream object, so there is no recursive assignment.

Reply:
I'm thinking about a simple destructuring instead, wdyt?

Reply:
Addresed in [a9f5ff9](https://github.com/elastic/kibana/pull/172462/commits/a9f5ff913ccf2f83ecc313b9f65eabe748b16a93).

Reply:
First thing i learned working with redux was immutability, and that Object.assign does shallow copy compared to merge which does deep copy